kotlin/display-grid/src/main/java/com/esri/arcgisruntime/sample/displaygrid/MainActivity.kt

```diff
@@ -74,184 +77,241 @@ class MainActivity : AppCompatActivity() {
-        adapter = ArrayAdapter(
-          this@MainActivity, android.R.layout.simple_spinner_item,
-          resources.getStringArray(R.array.colors_array)
-        ).also { labelColorAdapter ->
-          labelColorAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
-        }
+  /**
+   * Sets up the spinner for selecting a label color and handles behavior for when a new label color is selected.
+   *
+   * @param popupView the popup view inflated in onCreate()
+   */
+  private fun setupLabelColorSpinner(popupView: View) {
+    popupView.label_color_spinner.apply {
+      adapter = ArrayAdapter(
+        this@MainActivity, android.R.layout.simple_spinner_item,
+        resources.getStringArray(R.array.colors_array)
+      ).also { labelColorAdapter ->
+        labelColorAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+      }
 
-        // change grid labels color
-        onItemSelectedListener = object : OnItemSelectedListener {
-          override fun onItemSelected(
-            parent: AdapterView<*>?,
-            view: View,
-            position: Int,
-            id: Long
-          ) {
-            // set the color
-            when (position) {
-              0 -> labelColor = Color.RED
-              1 -> labelColor = Color.WHITE
-              2 -> labelColor = Color.BLUE
-              else -> Toast.makeText(
-                this@MainActivity,
-                "Unsupported option",
-                Toast.LENGTH_SHORT
-              ).show()
-            }
-            changeLabelColor(labelColor)
+      // change grid labels color
+      onItemSelectedListener = object : OnItemSelectedListener {
+        override fun onItemSelected(
+          parent: AdapterView<*>?,
+          view: View,
+          position: Int,
+          id: Long
+        ) {
+          // set the color
+          when (position) {
+            0 -> labelColor = Color.RED
+            1 -> labelColor = Color.WHITE
+            2 -> labelColor = Color.BLUE
+            else -> Toast.makeText(
+              this@MainActivity,
+              "Unsupported option",
+              Toast.LENGTH_SHORT
+            ).show()
           }
-
-          override fun onNothingSelected(parent: AdapterView<*>?) {}
+          changeLabelColor(labelColor)
         }
+
+        override fun onNothingSelected(parent: AdapterView<*>?) {}
       }
+    }
+  }
 
-      // create drop-down list of different label positions
-      label_position_spinner.apply {
-        adapter = ArrayAdapter(
-          this@MainActivity, android.R.layout.simple_spinner_item,
-          resources.getStringArray(R.array.positions_array)
-        ).also { positionAdapter ->
-          positionAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
-        }
+  /**
+   * Sets up the spinner for selecting a label position relative to the grid and handles behavior for when a label position is selected.
```

Comment:
this description is quite over the 120 char guideline for line length: let's break it up into two lines :)

Reply:
Good catch!